### PR TITLE
[RN] Add ability to share the URL for a conference

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

File: react/features/share-room/actions.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ * @param  {string} room - Room identifier.
+ * @private
+ * @returns {string|null}
+ */
+function _getRoomUrl(conference, room) {
+    if (!conference || !room) {
+        return null;
+    }
+
+    return `https://${conference.connection.options.hosts.domain}/${room}`;
```
Comment (saghul) [inline react/features/share-room/actions.js]: @lyubomir I just realized this is not going to work with deployments which serve meet in a sub-directory, do you remember where we store that piece of information?

Comment (lyubomir) [inline react/features/share-room/actions.js]: Well... we don't store it. There's `windowLocationContextRoot` in features/app/components/App.web.js.

File: react/features/share-room/middleware.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ * @param {Dispatch} dispatch - The Redux dispatch function.
+ * @private
+ * @returns {void}
+ */
+function _shareRoom(roomURL: string, dispatch: Function) {
+    // TODO The following display/human-readable strings were submitted for
+    // review before i18n was introduces in react/. However, I reviewed it
+    // afterwards. Translate the display/human-readable strings.
+    const message = `Click the following link to join the meeting: ${roomURL}`;
+    const title = 'Jitsi Meet Conference';
```
Comment (lyubomir) [inline react/features/share-room/middleware.js]: The display/human-readable strings were submitted for review long before we integrated i18n in react/. Unfortunately, I reviewed this PR long after that. So we'll eventually have to translate them.

**Comment by lyubomir** [comment]
I wasn't sure whether we should move the share-room feature in react/mobile/. The current implementation is definitely mobile-specific. But we also have a room sharing dialog on Web... So we may argue both ways I guess.

**Comment by saghul** [comment]
@lyubomir I think the mobile version cannot reuse the dialogs implementation, because this one uses that underlying sheet, so I'm not sure how much code (if any, at all) we could share. We can definitely try!

File: react/features/share-room/middleware.js (hunk 2 above)
Comment (saghul) [inline react/features/share-room/middleware.js]: Oh, damn, I completely forgot about that when I rebased / adjusted this PR. 😢

**Pull request merged**